docs/source/package_definition.rst

```diff
@@ -113,6 +113,12 @@ implicit :attr:`this` object:
    Do not reference other early bound or late bound attributes in
    your early bound function. An error will be raised if you do.
 
+.. warning::
+   Certain package attributes cannot be accessed inside of an early bound function such as :attr:`this.root`.
```

Comment:
I would prefer is this was not a warning. Additionally, I would say something like "Note that the this object only exposes package attributes and nothing else when used inside an early function.", I feel like this better explains the situation.

We might also add such a note in the documentation of this (https://rez.readthedocs.io/en/stable/package_commands.html#this)? Though I don't know if any of the documented attributes in this are even exposed to early functions...

Reply:
@JeanChristopheMorinPerso I'm a bit confused by what you mean by "Note that the this object only exposes package attributes and nothing else when used inside an early function." Doesn't the `this` object always *only* expose package attributes? Or does it expose other attributes sometimes?

Reply:
Package attributes as in https://rez.readthedocs.io/en/stable/package_definition.html#package-attributes. That's different than what is available on https://rez.readthedocs.io/en/stable/package_commands.html#this.